src/averageAges.js

```diff
@@ -55,7 +77,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (onlyWithSon !== undefined)
+      ? people.find(woman => woman.name === person.mother)
+        && person.sex === 'm'
+      : people.find(woman => woman.name === person.mother)
+  );
+
+  return children.reduce((prev, current) =>
+    (prev + current.born - people.find(person =>
+      (person.name === current.mother)).born), 0) / children.length;
```

Comment:
Same) Checklist point 2: use parentheses

Reply:
i will fix it!